src/blocksparsearray/blocksparsearray.jl

```diff
@@ -1,168 +1,223 @@
-using BlockArrays: BlockArrays, Block, BlockedUnitRange, blockedrange, blocklength
+using BlockArrays:
+  BlockArrays,
+  Block,
+  BlockedUnitRange,
+  UndefBlocksInitializer,
+  blockedrange,
+  blocklength,
+  undef_blocks
 using DerivableInterfaces: @interface
 using Dictionaries: Dictionary
 using SparseArraysBase: SparseArrayDOK
 
-# TODO: Delete this.
-## using BlockArrays: blocks
+"""
+    SparseArrayDOK{T}(::UndefBlocksInitializer, axes)
+    SparseArrayDOK{T,N}(::UndefBlocksInitializer, axes)
```

Comment:
```suggestion
    SparseArrayDOK{T}(undef_blocks, axes)
    SparseArrayDOK{T,N}(undef_blocks, axes)
```

I think this is a bit more helpful, and typically also how `undef` is mentioned

Reply:
If necessary, we could also consider adding
```
See also: [`undef_blocks`](@ref BlockArrays.undef_blocks)`
```

Reply:
Good suggestion, I'll change that.

Reply:
It looks like you need some external tools to cross reference the docs of a function defined in another package so I just made a link manually for now.